modules/library/main/src/main/java/org/geotools/filter/visitor/DuplicatingFilterVisitor.java

```diff
@@ -339,14 +339,22 @@ public Object visit(NilExpression expression, Object extraData) {
     public Object visit(Add expression, Object extraData) {
         Expression expr1 = visit(expression.getExpression1(), extraData);
         Expression expr2 = visit(expression.getExpression2(), extraData);
-        return getFactory(extraData).add(expr1, expr2);
+        Add result = getFactory(extraData).add(expr1, expr2);
+        if (expr1 instanceof Literal && expr2 instanceof Literal) {
+            return getFactory(extraData).literal(result.evaluate(null));
```

Comment:
should a **duplicating** visitor be changing the expression? I'm happy for the simplifying visitor to do this but I think I'd expect to get back an identical filter with a duplication

Reply:
Whoops, you're right, this could should have been in the SimplifyingFilterVisitor. I guess that one did not have an ovverride and the IDE brought me to this class instead. Thanks for spotting it, fixing.